### PR TITLE
Use ci.cfg for ci and tox with active return-status-codes, fix #261

### LINKED ISSUE #261: No status-codes return value from code-analysis in tox tests
The tox tests check for the return value of the code-analysis output, but the current configuration disables those in the `buildout.cfg.bob` template. It should be `False` for development and `True`  for CI, but the tox configuration seems to be missing something. Example test: https://travis-ci.org/plone/bobtemplates.plone/jobs/346936640#L1701

### PR DESCRIPTION
**Pull request by MrTango**

### CODE REVIEW

File: skeleton-tests/test_addon.py [hunk 1]
```diff
@@ -89,12 +89,21 @@ def test_addon(tmpdir, capsys, config):
         )
         assert install_buildout_result == 0
         annotate_result = subprocess.call(
-            ['bin/buildout', 'annotate'],
+            [
+                'bin/buildout',
+                '-c',
```
Comment (loechel) [inline skeleton-tests/test_addon.py]: the buildout command should split into:

```python

[
    'bin/buildout',
    'code-analysis:return-status-codes=True',
    'annotate'
]

```

File: skeleton-tests/test_addon.py [hunk 2]
```diff
@@ -89,12 +89,21 @@ def test_addon(tmpdir, capsys, config):
...
+                'ci.cfg',
+                'annotate',
+            ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon.py]: corresponding:
the buildout command should split into:

```python

[
    'bin/buildout',
    'code-analysis:return-status-codes=True',
]

```

File: skeleton-tests/test_addon_behavior.py [hunk 3]
```diff
@@ -91,13 +91,19 @@ def test_behavior(tmpdir, capsys, config):
         annotate_result = subprocess.call(
             [
                 'bin/buildout',
+                '-c',
```
Comment (loechel) [inline skeleton-tests/test_addon_behavior.py]: see above

File: skeleton-tests/test_addon_behavior.py [hunk 4]
```diff
@@ -91,13 +91,19 @@ def test_behavior(tmpdir, capsys, config):
...
+                'ci.cfg',
                 'annotate',
             ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon_behavior.py]: see above

File: skeleton-tests/test_addon_content_type.py [hunk 5]
```diff
@@ -111,12 +111,21 @@ def test_addon_content_type(tmpdir, capsys, config):
         )
         assert install_buildout_result == 0
         annotate_result = subprocess.call(
-            ['bin/buildout', 'annotate'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon_content_type.py]: see above

File: skeleton-tests/test_addon_content_type.py [hunk 6]
```diff
@@ -111,12 +111,21 @@ def test_addon_content_type(tmpdir, capsys, config):
...
+                'ci.cfg',
+                'annotate',
+            ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon_content_type.py]: see above

File: skeleton-tests/test_addon_theme.py [hunk 7]
```diff
@@ -83,12 +83,21 @@ def test_addon_theme(tmpdir, capsys, config):
         )
         assert install_buildout_result == 0
         annotate_result = subprocess.call(
-            ['bin/buildout', 'annotate'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon_theme.py]: see above

File: skeleton-tests/test_addon_theme.py [hunk 8]
```diff
@@ -83,12 +83,21 @@ def test_addon_theme(tmpdir, capsys, config):
...
+                'ci.cfg',
+                'annotate',
+            ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
```
Comment (loechel) [inline skeleton-tests/test_addon_theme.py]: see above

File: skeleton-tests/test_addon_vocabulary.py [hunk 9]
```diff
@@ -92,12 +92,18 @@ def test_vocabulary(tmpdir, capsys, config):
...
+                'annotate',
+            ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
+                '-c ci.cfg',
```
Comment (loechel) [inline skeleton-tests/test_addon_vocabulary.py]: see above

File: skeleton-tests/test_theme_package.py [hunk 10]
```diff
@@ -91,12 +91,21 @@ def test_theme_package(tmpdir, capsys, config):
         )
         assert install_buildout_result == 0
         annotate_result = subprocess.call(
-            ['bin/buildout', 'annotate'],
+            [
+                'bin/buildout',
+                '-c',
```
Comment (loechel) [inline skeleton-tests/test_theme_package.py]: see above

File: skeleton-tests/test_theme_package.py [hunk 11]
```diff
@@ -91,12 +91,21 @@ def test_theme_package(tmpdir, capsys, config):
...
+                'annotate',
+            ],
             cwd=wd,
         )
         assert annotate_result == 0
         buildout_result = subprocess.call(
-            ['bin/buildout'],
+            [
+                'bin/buildout',
+                '-c',
```
Comment (loechel) [inline skeleton-tests/test_theme_package.py]: see above

**Comment by loechel** [comment]
I am actually not sure if this pull request is necessary at all, as we always want to fail code-analysis on tests.

**Comment by MrTango** [comment]
thx @loechel for the review, i made the changes ;)

**Pull request merged**